src/include/simeng/RegisterValue.hh

```diff
@@ -90,11 +96,16 @@ class RegisterValue {
    * the specified datatype. */
   template <class T>
   const T* getAsVector() const {
-    static_assert(alignof(T) <= 8 && "Alignment over 8 bytes not guranteed");
+    static_assert(alignof(T) <= 8 && "Alignment over 8 bytes not guaranteed");
     assert(bytes > 0 && "Attempted to access an uninitialised RegisterValue");
-    assert(sizeof(T) <= bytes &&
-           "Attempted to access a RegisterValue as a datatype larger than the "
-           "data held");
+    assert((sizeof(T) <= bytes || (bytes == 4 && sizeof(T) == 8)) && "Attempted"
+           " to access a RegisterValue as a datatype larger than the "
+           "data held" );
+    if(!relaxedFor32bit_) { // maybe #ifdef if it makes slower?
+      assert(sizeof(T) <= bytes &&
```

Comment:
This will not work as assertions are disabled in the RELEASE build. So it will be a nop afaik.

Reply:
I don't understand why this has been added? Also, I think this is related to [ this](https://github.com/UoB-HPC/SimEng/pull/317/files#r1309922102) comment.